stl/inc/algorithm

```diff
@@ -5314,6 +5314,56 @@ _NODISCARD _FwdIt max_element(_ExPo&&, _FwdIt _First, _FwdIt _Last) noexcept /*
     // not parallelized at present, parallelism expected to be feasible in a future release
     return _STD max_element(_First, _Last);
 }
+
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::max_element
+    template <class _It, class _Se, class _Pr, class _Pj>
+    _NODISCARD constexpr _It _Max_element_unchecked(_It _First, const _Se _Last, _Pr _Pred, _Pj _Proj) {
+        _STL_INTERNAL_STATIC_ASSERT(forward_iterator<_It>);
+        _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+        _STL_INTERNAL_STATIC_ASSERT(indirect_strict_weak_order<_Pr, projected<_It, _Pj>>);
+
+        auto _Found = _First;
+        if (_First == _Last) {
+            return _Found;
+        }
+
+        while (++_First != _Last) {
+            if (_STD invoke(_Pred, _STD invoke(_Proj, *_Found), _STD invoke(_Proj, *_First))) {
+                _Found = _First;
+            }
+        }
+
+        return _Found;
+    }
+
+    class _Max_element_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <forward_iterator _It, sentinel_for<_It> _Se, class _Pj = identity,
+            indirect_strict_weak_order<projected<_It, _Pj>> _Pr = ranges::less>
+        _NODISCARD constexpr _It operator()(_It _First, _Se _Last, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+            _Seek_wrapped(_First, _RANGES _Max_element_unchecked(_Get_unwrapped(_STD move(_First)),

```

Comment:
> It's a little strange to have `_Ugly` names overloaded between `std` and `ranges`. 

Brace yourself for #930 - it's becoming commonplace to have identically named helpers with identical functionality modulo sentinels and projections. The practice was new enough in this PR that I hadn't fully developed the style yet. These started as private members of `_Meow_fn` classes, then moved to namespace scope when I realized I needed to call them from external code. I decided it was still unnecessary to qualify the calls since the `_RANGES` versions always have at least one more argument for a projection. By the time I got to the `heap` family it occurred to me that consistently qualifying makes it obvious which function we're calling with less need for context. I'll audit these.

> Would it be better to choose a unique name?

Identically-named helpers have identical functionality and nearly identical implementations, they almost certainly share bugs and should be maintained together. To me, `_RANGES _Max_element_unchecked` and `_STD _Max_element_unchecked` should have names as similar as do `_RANGES max_element` and `_STD max_element`. I suspect I'll be able to factor many/most of these back into single functions when I do my reintegration pass after finishing the initial implementation of Ranges.

`_Equal_rev_pred`/`_Equal_rev_pred_unchecked` is weird because - for reasons unknown - I renamed only the `_RANGES` overload after deciding it was weird to have a `_Meow_unchecked` with no corresponding checked `meow` / `_Meow`. This will probably also come out in the wash during reintegration.

Reply:
Thanks for the rationale for the identical names; consistent qualification addresses my concerns about ADL.